packages/babel-cli/src/babel/dir.ts

```diff
@@ -6,12 +6,14 @@ import * as util from "./util";
 import * as watcher from "./watcher";
 import type { CmdOptions } from "./options";
 
-const FILE_TYPE = Object.freeze({
-  NON_COMPILABLE: "NON_COMPILABLE",
-  COMPILED: "COMPILED",
-  IGNORED: "IGNORED",
-  ERR_COMPILATION: "ERR_COMPILATION",
-} as const);
+const enum FILE_TYPE {
+  NON_COMPILABLE = "NON_COMPILABLE",
+  COMPILED = "COMPILED",
+  IGNORED = "IGNORED",
+  ERR_COMPILATION = "ERR_COMPILATION",
+}
+
+const outputtedFiles = new Set<string>();
```

Comment:
I think we don't have to maintain a set of printed files, the input path `src` is already available in `write`, we can compare `src` against the output path `path.resolve(dest)` and throw an error there.

Reply:
That's how I did it at first, but then I realized it also depends on the extension. Even the same directory may work correctly with different extensions.

Reply:
In that case we can compare the file path + base name without extensions, so we will throw if `/foo/bar.jsx` is transpiled to `/foo/bar.js` in `--watch` mode.